Tests/MongoSwiftTests/TestUtils.swift

```diff
@@ -44,6 +44,11 @@ class MongoSwiftTestCase: XCTestCase {
     // Use MemoryLayout instead of Int.bitWidth to avoid a compiler warning.
     // See: https://forums.swift.org/t/how-can-i-condition-on-the-size-of-int/9080/4 */
     static let is32Bit = MemoryLayout<Int>.size == 4
+
+    /// Generates a unique collection name of the format "-[<Test Case> <Test Name>]-<timestamp>".
+    internal func generateCollectionName() -> String {
+        return "\(self.name)-\(Date().timeIntervalSinceReferenceDate)"
```

Comment:
Is there any risk of exceeding the 120-character limit when combining the database and collection name?

I'm not sure if a timestamp component on the collection name makes sense, as that means we no longer have deterministic collection names.

For some prior art, which inspired the original request for Swift, we do the following in PHP:

 * In the low-level extension, we use the relevant portion of the path (basically a logical group name) and the test filename (test name) and replace any special characters with underscores: https://github.com/mongodb/mongo-php-driver/blob/master/tests/utils/tools.php#L396
 * In the library, we use the short test name (e.g. test method) and append a short hash of the full name (method and class): https://github.com/mongodb/mongo-php-library/blob/master/tests/TestCase.php#L148

In both cases, we can trust that each test will use the same collection on successive test suite runs. When a test succeeds, we clean up the collection(s) under test, but leave them as-is after a failure so the data can be inspected if needed.

Reply:
The risk is definitely there, but I just didn't think it was likely enough to worry about. I originally didn't want to hash them because it would be harder to manually inspect after a failure, but your second method (combining the test name plus a hash) seems to fix that issue.

Also, I misunderstood what this ticket was going for when I added the timestamp. I included it because I thought we _wanted_ to have unique collection names for every run of the test (so a stopped test run wouldn't require cleanup and past failures wouldn't influence future attempts). 

Anyways, I think I will go with the naming scheme in the library (or something similar).

Reply:
I ended up doing basically what the low level extension did, except instead of using the path, I used the test case name (file name) and the test name (function name). It turns out that hashing strings in Swift does not have great standard library support, and I wouldn't be able to do it here without involving some ObjC stuff.

To distinguish between collections within the same test, I appended something to the end, usually keeping with what was there before (e.g. a number, or a filename plus number in the case of crud tests).